plugins/modules/ndo_dhcp_option_policy.py

```diff
@@ -227,11 +220,38 @@ def main():
         mso.existing = {}
 
     if not module.check_mode and ops:
-        mso.request(mso_template.template_path, method="PATCH", data=ops)
+        mso_template.template = mso.request(mso_template.template_path, method="PATCH", data=ops)
+        match = get_dhcp_option_policy(mso_template, option_policy_uuid, option_policy)
+        if match:
+            mso.existing = match.details  # When the state is present
+        else:
+            mso.existing = {}  # When the state is absent
+    elif module.check_mode and state != "query":  # When the state is present/absent with check mode
+        mso.existing = mso.proposed if state == "present" else {}
 
     mso.exit_json()
 
 
+def get_dhcp_option_policy(mso_template, uuid=None, name=None, fail_module=False):
```

Comment:
why not set this in template like u do for others?

Reply:
Currently, I do not see the use case for moving it to the template module.